internal/controllers/controllers.go

```diff
@@ -0,0 +1,83 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// This label is set on all dynamic objects, to limit caches.
+	DynamicCacheLabel = "package-operator.run/cache"
+	// Revision annotations holds a revision generation number to order ObjectSets.
+	RevisionAnnotation = "package-operator.run/revision"
+)
+
+func EnsureCommonFinalizer(
```

Comment:
Is this supposed to be a generic function? If so, I think the paramter `cacheFinalizer` should be changed to `finalizer`